contracts/WildcardSteward_v2.sol

```diff
@@ -419,77 +470,188 @@ contract WildcardSteward_v2 is Initializable {
-            address benefactor = benefactors[tokenId];
-            benefactorFunds[benefactor] = benefactorFunds[benefactor].add(
-                collection
-            );
 
-            emit CollectPatronage(
-                tokenId,
-                currentOwner,
-                deposit[currentOwner],
-                collection
-            );
+            _collectLoyaltyPatron(tokenPatron, timeSinceLastMint);
+            emit RemainingDepositUpdate(tokenPatron, deposit[tokenPatron]);
+        }
+    }
+
+    function safeSend(uint256 _wei, address recipient)
+        internal
+        returns (bool transferSuccess)
+    {
+        (transferSuccess, ) = recipient.call.gas(2300).value(_wei)("");
+    }
+
+    function _updateBenefactorBalance(address benefactor) internal {
+        uint256 unclaimedPayoutAvailable = unclaimedPayoutDueForOrganisation(
+            benefactor
+        );
+
+        if (unclaimedPayoutAvailable > 0) {
+            if (
+                unclaimedPayoutAvailable.add(benefactorFunds[benefactor]) <
+                benefactorCredit[benefactor]
+            ) {
+                // Here there is nothing left extra to pay the organisation, everything goes to paying of debt.
+                benefactorCredit[benefactor] = benefactorCredit[benefactor].sub(
+                    unclaimedPayoutAvailable.add(benefactorFunds[benefactor])
+                );
+            } else {
+                benefactorFunds[benefactor] = unclaimedPayoutAvailable
+                    .add(benefactorFunds[benefactor])
+                    .sub(benefactorCredit[benefactor]);
+
+                benefactorCredit[benefactor] = 0;
+            }
+        }
+    }
+
+    function withdrawBenefactorFundsTo(address payable benefactor) public {
+        require(
+            benefactorLastTimeCollected[benefactor].add(1 days) > now,
+            "Cannot call this function more than once a day"
+        );
+
+        _updateBenefactorBalance(benefactor);
+
+        uint256 availableToWithdraw = benefactorFunds[benefactor];
+
+        if (availableToWithdraw > 0) {
+            if (availableToWithdraw > globalBenefactorDailyWithdrawalLimit) {
+                if (
+                    safeSend(globalBenefactorDailyWithdrawalLimit, benefactor)
+                ) {
+                    benefactorFunds[benefactor] = availableToWithdraw.sub(
+                        globalBenefactorDailyWithdrawalLimit
+                    );
+                } else {
+                    benefactorFunds[benefactor] = availableToWithdraw;
+                }
+            } else {
+                if (
+                    safeSend(globalBenefactorDailyWithdrawalLimit, benefactor)
+                ) {
+                    benefactorFunds[benefactor] = 0;
+                } else {
+                    benefactorFunds[benefactor] = availableToWithdraw;
+                }
+            }
+        }
+    }
+
+    function withdrawBenefactorFundsToValidated(
```

Comment:
This function looks like its still a WIP

Reply:
Yup, basically we need to fix line 629. Shouldn't be much work to get going.